ui/components/component-library/form-text-field/form-text-field.types.ts

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import type { PolymorphicComponentPropWithRef } from '../box';
+import { TextFieldProps } from '../text-field/text-field.types';
+import type { LabelProps } from '../label/label.types';
+import type { HelpTextProps } from '../help-text/help-text.types';
+
+export enum FormTextFieldSize {
+  Sm = 'sm',
+  Md = 'md',
+  Lg = 'lg',
+}
+
+export interface FormTextFieldStyleUtilityProps
+  extends Omit<TextFieldProps<'input'>, 'size' | 'type'> {
+  /**
+   * An additional className to apply to the FormTextField
+   */
+  className?: string;
+  /**
+   * The size of the FormTextField
+   */
+  size?: FormTextFieldSize;
+  /**
+   * Label for the FormTextField
+   */
+  label?: string | React.ReactNode;
+  /**
+   * Props to be passed to the Label component
+   */
+  labelProps?: LabelProps<'label'>;
+  /**
+   * Props for the TextField component within the FormTextField
+   */
+  textFieldProps?: TextFieldProps<'input'>;
```

Comment:
I believe the wrapper that receives the spread props is still a div 
```suggestion
  textFieldProps?: TextFieldProps<'div'>;
```

Reply:
Indeed, it should. Here, the `input` in `ElementType` refers to all props spread to the wrapper element, which is a `div` in this case. Therefore, we should be accepting valid `div` attributes. The attributes specific to the `input` element are defined in `inputProps` as seen here: https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/component-library/text-field/text-field.types.ts#L61